src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/UuidBehavior.php

```diff
@@ -159,8 +159,15 @@ protected function prepareKeyStatement(string $prefix): string
                 ));
             }
 
-            $getter = sprintf('get%s()', $filter->filter($column));
-            $keyStatement .= sprintf(" . '.' . \$this->%s", $getter);
+            switch ($this->getTable()->getColumn($column)->getType()) {
```

Comment:
@odokienko `switch` is evil.

Reply:
Please add tests. Please cover especially time based column's
http://propelorm.org/documentation/reference/schema.html

Reply:
Done